src/Datasource/EntityTrait.php

```diff
@@ -787,6 +787,43 @@ public function getDirty()
     }
 
     /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property The field to get value for.
+     * @return mixed
+     * @throws \InvalidArgumentException if the property name doesn't exist
+     */
+    public function &getOrFail($property)
+    {
+        $value = $this->get($property);
+        if ($value === null && !$this->propertyExists($property)) {
+            throw new \InvalidArgumentException('Property "' . $property . '" not found.');
+        }
+
+        return $value;
+    }
+
+    /**
+     * Check if real or virtual property for existence
+     *
+     * @param string $property The field to check for existence.
+     * @return bool
+     */
+    public function propertyExists($property)
+    {
+        if (array_key_exists($property, $this->_properties)) {
```

Comment:
Are you certain that `array_key_exists` on the private `_properties` "property" vs `property_exists` is better?
Is the core as well?

Reply:
Actually no I am not sure but I think that this piece of code is very simple to read and therefore okay. I am open to a better solution ;-)

Reply:
@inoas how would you check if a key exists in an array using `property_exists`?

Reply:
@robertpustulka I am just worried if someone sets up a real property it is not getting returned by this despite accessible via `$entity->myCustomProperty`.

Reply:
@inoas would a real property still be a property from the ORM perspective?

Reply:
@robertpustulka should it be? Could it be? It would if both $entity->real_property and $entity->get('real_property') / $entity->set('real_property', $value) would allow access, right?

```php
        logdebug($cmsDocument->foo_bar_quux); // returns value
        logdebug($cmsDocument->get('foo_bar_quux')); // does not return value
```

So right now this is no requirement as it is not transparent to the ORM.